contrib/pg_tde/src/smgr/pg_tde_smgr.c

```diff
@@ -68,6 +60,17 @@ tde_smgr_get_key(SMgrRelation reln, RelFileLocator *old_locator, bool can_create
 		return rkd;
 	}
 
+	/* if this is a CREATE TABLE, we have to generate the key */
+	if (IsCatalogRelationOid(reln->smgr_rlocator.locator.relNumber) && can_create)
+	{
+		return pg_tde_create_smgr_key(&reln->smgr_rlocator.locator);
+	}
+
+	event = GetCurrentTdeCreateEvent();
+
+	/* TODO: Can this ever happen? */
```

Comment:
No, unless we have data corruption, this can't happen, we would crash in the next line. But maybe it's worth a sanity check and a proper error instead.

Reply:
Ok!